src/MercadoPago/Client/Invoice/InvoiceClient.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace MercadoPago\Client\Invoice;
+
+use MercadoPago\Client\MercadoPagoClient;
+use MercadoPago\Core\MPRequestOptions;
+use MercadoPago\Exceptions\MPApiException;
+use MercadoPago\MercadoPagoConfig;
+use MercadoPago\Net\HttpMethod;
+use MercadoPago\Net\MPSearchRequest;
+use MercadoPago\Resources\Invoice;
+use MercadoPago\Resources\InvoiceSearch;
+use MercadoPago\Serialization\Serializer;
+
+/** Client responsible for performing invoice actions. */
+class InvoiceClient extends MercadoPagoClient
+{
+    private static $URL_WITH_ID = "/authorized_payments/%s";
+
+    private static $URL_SEARCH = "/authorized_payments/search";
+
+    /** Default constructor. Uses the default http client used by the SDK. */
+    public function __construct()
+    {
+        parent::__construct(MercadoPagoConfig::getHttpClient());
+    }
+
+    /**
+     * Method responsible for getting an invoice.
+     * @param int $id invoice id.
+     * @param mixed $request_options request options to be sent.
+     * @return \MercadoPago\Resources\Invoice invoice found.
+     */
+    public function get(int $id, ?MPRequestOptions $request_options = null): Invoice
+    {
+        try {
+            $response = parent::send(sprintf(self::$URL_WITH_ID, $id), HttpMethod::GET, null, null, $request_options);
+            $result = Serializer::deserializeFromJson(Invoice::class, $response->getContent());
+            $result->setResponse($response);
+            return $result;
+        } catch (MPApiException | \Exception $e) {
+            throw $e;
```

Comment:
What is the intended behavior here?

Reply:
If we receive an API error, a treatment is made so that the integrator has access to more information in a more friendly way with the exception of `MPApiException`. Any other errors will be generalized as `\Exception`. Both are threw for the integrator to handle however they prefer.

Reply:
I don't understand.
There is no treatment here, so the same behavior is being produced regardless the exception matches the types defined in the `catch` block or if they are thrown without a previous match.

Reply:
In this function, which is responsible for making HTTP calls, we handle errors as follows: If an error occurs when making the call, we will return an `Exception` with the error in a more generic way. But if it is a call with status code `< 200 or >= 300`, we create this `MPApiException` exception. Both are thrown to clients, and clients should not make any treatment, just throw to the integrator's application. It is the responsibility of the integrator to handle these exceptions.

Reply:
What I mean is, in the current version of this code, you're throwing the same exception, regardless it was previously caught by the `catch` block or not:
* If the exception is of type `MPApiException` or `\Exception`, the `catch` block is throwing the very same exception, without any treatment;
* If the exception is not of type `MPApiException` or `\Exception`, the exception bubbles up and is thrown as is because there is no `catch` block processing it.

BTW, there is no sense in `MPApiException | \Exception`, as `MPApiException` already extends from `\Exception`. In other words, `MPApiException` is narrowing nothing there:
https://github.com/mercadopago/sdk-php/blob/9675436df94c9c59f9721a366c9a598612b4a411/src/MercadoPago/Exceptions/MPApiException.php#L9

Reply:
Right, but in this case, if we only catch `\Exception`, the integrator will not be able to access the `$status_code` and `$api_response` properties that are inside `MPApiResponse`, so we throw both `MPApiException` and `\Exception` to also be throwing any others exceptions. In this case, as we just throw exceptions without any additional handling, do you mean that we don't need to do this `catch`? Will exceptions be thrown automatically? Can you give an example of what the correction would look like?